services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -530,6 +531,10 @@ public Response createLoginPassword(){
 
     @Override
     public Response createPasswordReset() {
+        MultivaluedMap<String, String> params = context.getUriInfo().getQueryParameters();
```

Comment:
I think that the "login_hint" can be rather obtained from the authentication session with something like:
```
authenticationSession.getClientNote(OIDCLoginProtocol.LOGIN_HINT_PARAM)
```
This will be better than reading it directly from request parameters, as request parameters are OIDC-protocol specific.